### PR TITLE
WEBITOOLS-231: Configs Fastscan for v3

### PR DESCRIPTION
**Pull request by CAST-ANJ**
Do not merge

### CODE REVIEW

File: aip-console-tools-cli/src/test/java/com/castsoftware/aip/console/tools/OnboardApplicationFastScanCommandIntegrationTest.java [hunk 1]
```diff
@@ -57,7 +57,7 @@
 @ActiveProfiles(TestConstants.PROFILE_INTEGRATION_TEST)
 public class OnboardApplicationFastScanCommandIntegrationTest extends AipConsoleToolsCliBaseTest {
```
Comment (leon2Meudon92) [inline aip-console-tools-cli/src/test/java/com/castsoftware/aip/console/tools/OnboardApplicationFastScanCommandIntegrationTest.java]: Also rename https://cast-products.atlassian.net/browse/WEBI-16170 to FastScanIntegrationTes
See OnbordApplication if such name already exist

File: aip-console-tools-core/src/main/java/com/castsoftware/aip/console/tools/core/dto/ApplicationCommonDetails.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+import lombok.Getter;
+import lombok.Setter;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Getter
+@Setter
+public class ApplicationCommonDetails {
```
Comment (leon2Meudon92) [inline aip-console-tools-core/src/main/java/com/castsoftware/aip/console/tools/core/dto/ApplicationCommonDetails.java:11]: No meaning in having dto with just one member
Use Builder.Default

File: aip-console-tools-cli/src/main/java/com/castsoftware/aip/console/tools/commands/FasctScanCommand.java [hunk 3]
```diff
@@ -61,7 +62,7 @@ public class OnboardApplicationFastScanCommand extends BasicCallable {
     //This version can be null if failed to convert from string
     private static final VersionInformation MIN_VERSION = VersionInformation.fromVersionString("2.8.0");
 
-    public OnboardApplicationFastScanCommand(RestApiService restApiService, JobsService jobsService, UploadService uploadService, ApplicationService applicationService) {
+    public FasctScanCommand(RestApiService restApiService, JobsService jobsService, UploadService uploadService, ApplicationService applicationService) {
         super(restApiService, jobsService, uploadService, applicationService);
```
Comment (leon2Meudon92) [inline aip-console-tools-cli/src/main/java/com/castsoftware/aip/console/tools/commands/FasctScanCommand.java]: Please try unsing *astScan(FastScanProperties fastScanProperties)* everywhere fastscan needed

File: aip-console-tools-core/src/main/java/com/castsoftware/aip/console/tools/core/dto/ApplicationCommonDetails.java (hunk 2 above)
Comment (CAST-ANJ) [reply aip-console-tools-core/src/main/java/com/castsoftware/aip/console/tools/core/dto/ApplicationCommonDetails.java:11]: These details might increase going down the line. So just kept it for now.

File: aip-console-tools-cli/src/test/java/com/castsoftware/aip/console/tools/OnboardApplicationFastScanCommandIntegrationTest.java (hunk 1 above)
Comment (CAST-ANJ) [reply aip-console-tools-cli/src/test/java/com/castsoftware/aip/console/tools/OnboardApplicationFastScanCommandIntegrationTest.java]: Changes done

File: aip-console-tools-cli/src/main/java/com/castsoftware/aip/console/tools/commands/FasctScanCommand.java (hunk 3 above)
Comment (CAST-ANJ) [reply aip-console-tools-cli/src/main/java/com/castsoftware/aip/console/tools/commands/FasctScanCommand.java]: Changes done

**Comment by leon2Meudon92** [comment]
Create application is not required.
Just FastScan, DeepAnalyze and OnboardApplication

File: aip-console-jenkins/src/main/java/io/jenkins/plugins/aipconsole/FastScanAction.java [hunk 4]
```diff
@@ -9,8 +9,8 @@
 import javax.annotation.CheckForNull;
 import javax.annotation.Nullable;
 
-public class OnboardApplicationFastScanAction extends BaseAction implements RunAction2 {
-    private static final Logger LOGGER = LoggerFactory.getLogger(OnboardApplicationFastScanAction.class);
+public class FastScanAction extends BaseAction implements RunAction2 {
```
Comment (leon2Meudon92) [inline aip-console-jenkins/src/main/java/io/jenkins/plugins/aipconsole/FastScanAction.java]: Renaming action cases jobs based on PIPE failed.
So keep the same name for Jenkins and Give an alias if you want for CLI

File: aip-console-tools-core/src/main/java/com/castsoftware/aip/console/tools/core/dto/ApplicationCommonDetails.java [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+
+    public List<ApplicationCommonDetailsDto> applicationCommonDetailsDtoSet = new ArrayList<>();
```
Comment (leon2Meudon92) [inline aip-console-tools-core/src/main/java/com/castsoftware/aip/console/tools/core/dto/ApplicationCommonDetails.java:13]: Use Builder.Default

File: aip-console-tools-core/src/main/java/com/castsoftware/aip/console/tools/core/services/ApplicationServiceImpl.java [hunk 6]
```diff
@@ -233,8 +227,8 @@ public int publishToImaging(String applicationName, long sleepDuration, boolean
             }
 
             applicationDto = getApplicationDetails(applicationDto.getGuid());
-            Set<String> statuses = EnumSet.of(VersionStatus.ANALYSIS_DATA_PREPARED, VersionStatus.IMAGING_PROCESSED,
-                            VersionStatus.SNAPSHOT_DONE, VersionStatus.FULLY_ANALYZED, VersionStatus.ANALYZED).stream()
+            Set<String> statuses = Stream.of(VersionStatus.ANALYSIS_DATA_PREPARED, VersionStatus.IMAGING_PROCESSED,
```
Comment (leon2Meudon92) [inline aip-console-tools-core/src/main/java/com/castsoftware/aip/console/tools/core/services/ApplicationServiceImpl.java]: Keep enumSet

File: aip-console-tools-core/src/main/java/com/castsoftware/aip/console/tools/core/dto/ApplicationCommonDetails.java (hunk 5 above)
Comment (CAST-ANJ) [reply aip-console-tools-core/src/main/java/com/castsoftware/aip/console/tools/core/dto/ApplicationCommonDetails.java:13]: Why is builder.default needed over here?
I have added the builder annotation in the ApplicationCommonDetailsDto

File: aip-console-tools-core/src/main/java/com/castsoftware/aip/console/tools/core/services/ApplicationServiceImpl.java (hunk 6 above)
Comment (CAST-ANJ) [reply aip-console-tools-core/src/main/java/com/castsoftware/aip/console/tools/core/services/ApplicationServiceImpl.java]: Changes done

File: aip-console-jenkins/src/main/java/io/jenkins/plugins/aipconsole/FastScanAction.java (hunk 4 above)
Comment (CAST-ANJ) [reply aip-console-jenkins/src/main/java/io/jenkins/plugins/aipconsole/FastScanAction.java]: Changes done

File: aip-console-tools-core/src/main/java/com/castsoftware/aip/console/tools/core/dto/ApplicationCommonDetails.java (hunk 5 above)
Comment (CAST-ANJ) [reply aip-console-tools-core/src/main/java/com/castsoftware/aip/console/tools/core/dto/ApplicationCommonDetails.java:13]: Are we doing this for immutability and to be thread safe?

**Comment by leon2Meudon92** [comment]
You should update Console min version targetted by the tools
Should be >= 3.0.0

**Review by leon2Meudon92 — COMMENTED** [review]
You will be asked to group all commands together
Should have something like (Jenkins)
Run DeepAnalaysis
Run Schan Application
Run Onboard application

**Pull request merged**